src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -573,17 +576,33 @@ internal static bool Dispatch()
                         {
                             ThreadPool.ReportThreadStatus(isWorking: true);
                             reportedStatus = true;
-                            workItem.ExecuteWorkItem();
+                            if (workItem is IThreadPoolWorkItem wi)
```

Comment:
Would checking its a `Task` first be faster than checking if it implements the interface `IThreadPoolWorkItem`? (as interface lookup is a search which depends on order of declaration https://github.com/dotnet/corefx/issues/29158#issuecomment-382251806)